### ISSUE TITLE
Schema::Connection()  doesn't seem to work.

### ISSUE THREAD

**Issue by timothymarois** [laravel/framework#26385]
- Laravel Version: 5.7
- PHP Version: 7.1
- Database Driver & Version: MySQL

### Description:

Using Schema to change Connection to another MySQL database, does not work. Still injected the tables into the default database. 

This has no effect so far...
`Schema::connection('db2')`

Inside the database migration:
```php
Schema::connection('db2')->create('pages', function (Blueprint $table) {
            $table->increments('id');
            $table->string('name',64)->unique();
            $table->string('meta_description',155);
            $table->string('page_description',255);

            $table->timestamps();
            $table->softDeletes();
});

Schema::connection('db2')->dropIfExists('pages');
```

I have 2 databases, one for main (default) which has user details and the migration table. The second for specific project details. When the migration runs, it adds everything into db1 (default) even though I have specifically told it to use db2 connection. (if the connection is renamed to something that does not exist, it throws an error, so it is looking at it, but it is not using it)

There might be something I am not doing correctly, but nonetheless, I am using exactly what the docs have told me, so far does not work as intended. 

https://laravel.com/docs/5.7/migrations#tables

Inside `.env`

```
DB_HOST=127.0.0.1
DB_PORT=3306
DB_DATABASE=db_manager
DB_USERNAME=root
DB_PASSWORD=root
```

```
DB_2_HOST=127.0.0.1
DB_2_PORT=3306
DB_2_DATABASE=db_content
DB_2_USERNAME=root
DB_2_PASSWORD=root
```
 
Inside `database.php` 

`'default' => env('DB_CONNECTION', 'db1'),`

```
'connections' => [

        'db1' => [
            'driver' => 'mysql',
            'host' => env('DB_HOST', '127.0.0.1'),
            'port' => env('DB_PORT', '3306'),
            'database' => env('DB_DATABASE', 'forge'),
            'username' => env('DB_USERNAME', 'forge'),
            'password' => env('DB_PASSWORD', ''),
            'unix_socket' => env('DB_SOCKET', ''),
            'charset' => 'utf8mb4',
            'collation' => 'utf8mb4_unicode_ci',
            'prefix' => '',
            'strict' => true,
            'engine' => null,
        ],

        'db2' => [
            'driver' => 'mysql',
            'host' => env('DB_2_HOST', '127.0.0.1'),
            'port' => env('DB_2_PORT', '3306'),
            'database' => env('DB_2_DATABASE', 'forge'),
            'username' => env('DB_2_USERNAME', 'forge'),
            'password' => env('DB_2_PASSWORD', ''),
            'unix_socket' => env('DB_2_SOCKET', ''),
            'charset' => 'utf8mb4',
            'collation' => 'utf8mb4_unicode_ci',
            'prefix' => '',
            'strict' => true,
            'engine' => null,
        ],
];
```

The database connections are the same host, they just use different databases (but in the future the hosts could change, but that should not affect this from ignoring connection usage) 

Running a fresh migration:
`php artisan migrate:refresh --seed`

**Comment by staudenmeir** [comment]
I can't reproduce this, `Schema::connection()` works for me.

I don't see what you could be doing differently. Does the `db2` connection work when you retrieve data?

**Comment by timothymarois** [comment]
@staudenmeir Thanks for taking a look. I've just started to get into the data retrieval process, and looks like its doing the same thing. (looks more like Laravel-wide from my setup) - I have added the property in the model to load only the connection.

`protected $connection = 'db2';`

but it does print out the error: (when running a query)

`Base table or view not found: 1146 Table 'db_content. pages' doesn't exist `

Is there something in the laravel config that is preventing multiple connections (and of the same host) from loading up, because it looks to only be looking at the default database. Looking at the connection and configuration details above, why else would this not work?
